platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationContainer.java

```diff
@@ -300,102 +305,117 @@ public NamedDomainObjectProvider<DependencyScopeConfiguration> dependencyScope(S
-        return createLegacyConfiguration(name, ConfigurationRoles.DEPENDENCY_SCOPE, Actions.doNothing());
+    public Configuration dependencyScopeLocked(String name) {
+        assertCanMutate("dependencyScopeLocked(String)");
+        DefaultConfiguration result = createLegacyConfiguration(name, ConfigurationRoles.DEPENDENCY_SCOPE, Actions.doNothing());
+        result.preventUsageMutation();
+        return result;
     }
 
     @Override
-    public Configuration dependencyScopeUnlocked(String name, Action<? super Configuration> action) {
-        assertCanMutate("dependencyScopeUnlocked(String, Action)");
-        return createLegacyConfiguration(name, ConfigurationRoles.DEPENDENCY_SCOPE, action);
+    public Configuration dependencyScopeLocked(String name, Action<? super Configuration> action) {
+        assertCanMutate("dependencyScopeLocked(String, Action)");
+        DefaultConfiguration result = createLegacyConfiguration(name, ConfigurationRoles.DEPENDENCY_SCOPE, action);
+        result.preventUsageMutation();
+        return result;
     }
 
     @Override
     @Deprecated
-    public Configuration resolvableDependencyScopeUnlocked(String name) {
-        assertCanMutate("resolvableDependencyScopeUnlocked(String)");
-        return createLegacyConfiguration(name, ConfigurationRoles.RESOLVABLE_DEPENDENCY_SCOPE, Actions.doNothing());
+    public Configuration resolvableDependencyScopeLocked(String name) {
+        assertCanMutate("resolvableDependencyScopeLocked(String)");
+        DefaultConfiguration result = createLegacyConfiguration(name, ConfigurationRoles.RESOLVABLE_DEPENDENCY_SCOPE, Actions.doNothing());
+        result.preventUsageMutation();
+        return result;
     }
 
     @Override
     @Deprecated
-    public Configuration resolvableDependencyScopeUnlocked(String name, Action<? super Configuration> action) {
-        assertCanMutate("resolvableDependencyScopeUnlocked(String, Action)");
-        return createLegacyConfiguration(name, ConfigurationRoles.RESOLVABLE_DEPENDENCY_SCOPE, action);
+    public Configuration resolvableDependencyScopeLocked(String name, Action<? super Configuration> action) {
+        assertCanMutate("resolvableDependencyScopeLocked(String, Action)");
+        DefaultConfiguration result = createLegacyConfiguration(name, ConfigurationRoles.RESOLVABLE_DEPENDENCY_SCOPE, action);
+        result.preventUsageMutation();
+        return result;
     }
 
     @Override
-    public Configuration migratingUnlocked(String name, ConfigurationRole role) {
-        assertCanMutate("migratingUnlocked(String, ConfigurationRole)");
-        return migratingUnlocked(name, role, Actions.doNothing());
+    public Configuration migratingLocked(String name, ConfigurationRole role) {
+        assertCanMutate("migratingLocked(String, ConfigurationRole)");
+        DefaultConfiguration result = migratingLocked(name, role, Actions.doNothing());
+        result.preventUsageMutation();
+        return result;
```

Comment:
Shouldn't need to call `preventUsageMutation` on a configuration returned by `migratingLocked`, as it should already return a locked configuration.